repose-aggregator/services/service-client/impl/src/main/java/org/openrepose/core/services/serviceclient/akka/impl/AkkaServiceClientImpl.java

```diff
@@ -72,27 +74,42 @@ public AkkaServiceClientImpl(HttpClientService httpClientService) {
                 .expireAfterWrite(FUTURE_CACHE_TTL, FUTURE_CACHE_UNIT)
                 .build();
 
-
-        tokenActorRef = actorSystem.actorOf(new Props(new UntypedActorFactory() {
-            public UntypedActor create() {
-
-                return new AuthTokenFutureActor(serviceClient);
-            }
-        }).withRouter(new RoundRobinRouter(numberOfActors)), "authRequestRouter");
+        createTokenActor();
     }
 
-
     @PreDestroy
     public void destroy() {
-        //Tie this actor system to our bean lifecycle
+        // tie this actor system to our bean lifecycle
         actorSystem.shutdown();
     }
 
     @Override
+    public void setConnectionPoolId(String connectionPoolId) {
+        if (StringUtils.equals(this.connectionPoolId, connectionPoolId)) {
```

Comment:
This state should live in the actor, not in here. You should let the actors decide if they need to change their connection pool ID or not.

Getting sent multiple messages asking them to change their connection pool ID shouldn't matter.